gems/pending/util/extensions/miq-file.rb

```diff
@@ -53,12 +53,17 @@ def self.sizeEx(path)
 
   def self.path_to_uri(file, hostname = nil)
     hostname ||= MiqSockUtil.getFullyQualifiedDomainName
+    file = Addressable::URI.encode(file.tr('\\', '/'))
 
-    URI::Generic.build(
-      :scheme => 'file',
-      :host   => hostname,
-      :path   => "/#{Addressable::URI.encode(file.tr('\\', '/'))}"
-    ).to_s
+    begin
+      URI::Generic.build(
+        :scheme => 'file',
+        :host   => hostname,
+        :path   => "/#{file}"
+      ).to_s
+    rescue URI::InvalidComponentError # Punt on Windows volume names, etc
+      "file://#{hostname}/#{file}"
```

Comment:
I think it's still possible for `build` to raise a legitimate `InvalidComponentError`.  For instance, if it can't parse the `path`.

Is there anything you can use out of the `InvalidComponentError` to determine that you can assume that `"file://#{hostname}/#{file}"` is the correct result?